File: Client (Angular)/models/visit-page.ts
import {Animal, AnimalEntity} from './animal';
import {Visit, VisitEntity} from './visit';
import {Owner, OwnerEntity} from './owner';
import {PreventiveTreatment} from './preventive-treatment';
import {VisitPageLists, VisitPageListsEntity} from './visit-page-lists';

export interface VisitPage {
  visit: Visit;
  animal: Animal;
  owner: Owner;
  lists: VisitPageLists;
  preventiveTreatmentsList: PreventiveTreatment[];
  prescriptionsNumber: number;
}

export class VisitPageEntity implements VisitPage{
  constructor() {
    this.animal = new AnimalEntity();
    this.lists = new VisitPageListsEntity();
    this.owner = new OwnerEntity();
    this.preventiveTreatmentsList = [];
    this.visit = new VisitEntity();
    this.prescriptionsNumber = 0;
  }
  animal: Animal;
  lists: VisitPageLists;
  owner: Owner;
  preventiveTreatmentsList: PreventiveTreatment[] = [];
  visit: Visit;
  prescriptionsNumber: number;
}

File: Server (ASP.NET Core)/Petadmin/Models/VisitPage.cs
using System.Collections.Generic;
using Petadmin.Core.Models;
using PetAdmin.Core.Models;

namespace Petadmin.Models
{
    public class VisitPage
    {
        public Visit Visit { get; set; }
        public Animal Animal { get; set; }
        public Owner Owner { get; set; }
        public List<PreventiveTreatment> PreventiveTreatmentsList { get; set; }
        public VisitPageLists Lists { get; set; }
        public int PrescriptionsNumber  { get; set; }
    }
}
